WordPress/gutenberg
# Bring implementations of convert nav menu items to blocks back into sync

## Description
It looks like recently there was a regression whereby the matching implementations of `menuItemToBlockAttributes` in the Nav editor and block got out of sync.

This PR updates both sides and brings everything into sync following mine and @gwwar 's changes in the block and editor.

The fact that this happens points even more strongly towards the need for https://github.com/WordPress/gutenberg/issues/31580 to land.

## How has this been tested?
Check unit tests pass and both implementations still work correctly:

1. Check a menu can be viewed in the Editor.
2. Check a block can be created from menus.

## Screenshots 

## Types of changes





## Checklist:
